### PR TITLE
6442919: JFilechooser popup still left-to-right when JFilechooser is set to right-to-left

### PR DESCRIPTION
**Pull request by TejeshR13**
Popup menu is created on mouse click, hence the orientation for FileChooser won't apply to the popup menu created. Hence the fix address the issue by applying the FileChooser orientation when it is created on mouse operation (Right Click). 
The test does not apply to Motif and Aqua Look And Feel since right click is not supported on filePane of FileChooser, hence it is skipped. The fix is tested in CI and it is green.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-6442919](https://bugs.openjdk.org/browse/JDK-6442919): JFilechooser popup still left-to-right when JFilechooser is set to right-to-left (**Bug** - P4)

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14949/head:pull/14949` \
`$ git checkout pull/14949`

Update a local copy of the PR: \
`$ git checkout pull/14949` \
`$ git pull https://git.openjdk.org/jdk.git pull/14949/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14949`

View PR using the GUI difftool: \
`$ git pr show -t 14949`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14949.diff">https://git.openjdk.org/jdk/pull/14949.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14949#issuecomment-1643616228)

### CODE REVIEW

File: test/jdk/javax/swing/JFileChooser/FCPopupMenuOrientationTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+public class FCPopupMenuOrientationTest {
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeAndWait(new Runnable() {
+            @Override
+            public void run() {
+                for (UIManager.LookAndFeelInfo laf :
+                        UIManager.getInstalledLookAndFeels()) {
+                    String className = laf.getName().toLowerCase();
+                    if (className.contains("motif") || className.contains("mac")) {
```
Comment (mrserb) [inline test/jdk/javax/swing/JFileChooser/FCPopupMenuOrientationTest.java:51]: The test does not have a headful keyword so in the CI it was tested on a headless system where the GTK l&f is unsupported. Please double-check - do we need to skip it here or not?

Comment (TejeshR13) [reply test/jdk/javax/swing/JFileChooser/FCPopupMenuOrientationTest.java:51]: Yeah, I can see from log that it is unsupported. I will skip it.

Comment (TejeshR13) [reply test/jdk/javax/swing/JFileChooser/FCPopupMenuOrientationTest.java]: The fix doesn't apply to GTK too, thanks for pointing it out.

File: src/java.desktop/share/classes/sun/swing/FilePane.java [hunk 2]
```diff
@@ -1932,6 +1932,8 @@ public JPopupMenu getComponentPopupMenu() {
         if (viewMenu != null) {
             viewMenu.getPopupMenu().setInvoker(viewMenu);
         }
+
+        contextMenu.applyComponentOrientation(getFileChooser().getComponentOrientation());
```
Comment (DamonGuy) [inline src/java.desktop/share/classes/sun/swing/FilePane.java:1936]: I see that this bug is old. Seems like a simple way to fix this and I see similar calls for other UI elements, like JMenu. It looks like in JMenu, it iterates through all components and sets it orientation, but it looks like the popup is the only problem, so looks fine.

You still need to update the year in the header of FilePane.

File: test/jdk/javax/swing/JFileChooser/FCPopupMenuOrientationTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                    }
+                }
+            }
+        });
+        System.out.println("Test Pass");
+    }
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+            System.out.println("Set L&F : "+laf.getName());
```
Comment (DamonGuy) [inline test/jdk/javax/swing/JFileChooser/FCPopupMenuOrientationTest.java:76]:
```suggestion
            System.out.println("Set L&F : " + laf.getName());
```

File: src/java.desktop/share/classes/sun/swing/FilePane.java (hunk 2 above)
Comment (TejeshR13) [reply src/java.desktop/share/classes/sun/swing/FilePane.java:1936]: Yeah, `applyComponentOrientation ` iterates through all the child components and sets the orientation.

**Review by mrserb — APPROVED** [review]

**Review by kumarabhi006 — APPROVED** [review]

**Pull request closed without merging**